### ISSUE TITLE
`UnknownOperation` string representation should be `UnknownOperationException`

### ISSUE THREAD

**Issue by david-perez** [awslabs/smithy-rs#1430]
https://github.com/awslabs/smithy-rs/blob/c6d6c2b131259ecdbd9bd923a0a04db8c610854f/rust-runtime/aws-smithy-http-server/src/runtime_error.rs#L45-L53

The string representation is `UnknownOperationException`. See [the TypeScript implementation](https://github.com/awslabs/smithy-typescript/blob/main/smithy-typescript-codegen/src/main/resources/software/amazon/smithy/typescript/codegen/framework-errors.smithy).

I guess we didn't catch this because there aren't any protocol tests. But even if they existed, we wouldn't be able to pass them, since this error arises when the `Router` is not able to route to an operation, a code path we do not exercise because of https://github.com/awslabs/smithy-rs/issues/1212.

So once we do https://github.com/awslabs/smithy-rs/issues/1212, we should add some ambiguous routing protocol tests.

**Comment by 82marbag** [comment]
Fixed the string in https://github.com/awslabs/smithy-rs/pull/1424, keeping this open to add tests

**Comment by crisidev** [comment]
I think we can close this right?

**Comment by david-perez** [comment]
No, we're blocked on #1212 to add protocol tests to not regress on this.

> So once we do https://github.com/awslabs/smithy-rs/issues/1212, we should add some ambiguous routing protocol tests.
